File: src/main/java/aoc2019/day07/LoopAmplifierOutput.java
package aoc2019.day07;

import java.util.ArrayList;

public class LoopAmplifierOutput {

    private int outputValue;
    private int lastIndex;
    private ArrayList<Integer> currentNumbers;

    LoopAmplifierOutput(int outputValue, int lastIndex, ArrayList<Integer> numbers) {
        this.outputValue = outputValue;
        this.lastIndex = lastIndex;
        this.currentNumbers = numbers;
    }

    int getOutputValue() {
        return outputValue;
    }

    int getLastIndex() {
        return lastIndex;
    }

    public ArrayList<Integer> getNumbers() {
        return currentNumbers;
    }

    @Override
    public String toString() {
        return "{" +
                "outputValue=" + outputValue +
                ", lastIndex=" + lastIndex +
                ", currentNumbers=" + currentNumbers +
                '}';
    }
}

File: src/main/kotlin/aoc2019/day07/LoopAmplifierOutput.kt
package aoc2019.day07

import java.util.ArrayList

class LoopAmplifierOutput internal constructor(val outputValue: Int, val lastIndex: Int, val numbers: ArrayList<Int>) {
    override fun toString(): String {
        return "{" +
                "outputValue=" + outputValue +
                ", lastIndex=" + lastIndex +
                ", currentNumbers=" + numbers +
                '}'
    }
}